backend/src/supabase/functions/get-room-permission/index.ts

```diff
@@ -0,0 +1,60 @@
+import { serve } from 'https://deno.land/std@0.131.0/http/server.ts';
+import { createClient } from 'https://esm.sh/@supabase/supabase-js@2';
+import { corsHeaders } from '../_shared/cors.ts';
+
+async function getPermission(supabaseClient, body) {
+  let { userId, roomId }= body;
+  roomId = parseInt(roomId);
+  const roomsData = await supabaseClient.from('rooms-data').select('permissionId').match({ roomId,  userId });
+  if (roomsData.data && roomsData.data.length === 0) {
+    return new Response(JSON.stringify({
+      error: 'No room or user with given data exists'
+    }), {
+      headers: {
+        ...corsHeaders,
+        'Content-Type': 'application/json'
+      },
+      status: 404
+    });
+  }
```

Comment:
Question: which is the best way to handle this case? If we have no result here, it just means that this user has no special permissions in this room: it's not technically an error.

If we send a 404 error here, we'll likely need to handle it separately on the Front-End. Alternatively, we could send a 200 response with `permission: null`. Then, on the Front-End, we'll set up a logic like `if (!res.data.permission) { // user is guest }`.

What do you think?

Reply:
I agree with @marcovidonis here. 404 should be returned if for example the user or the room don't exist to begin with. But the user having no roles is an expected possible result that should return 200